src/api.ts

```diff
@@ -1,103 +1,188 @@
-import axios from "axios";
-import { get, sleep } from "./common";
-import type { Vultr } from "./vultr";
+import { sleep } from "./common";
+import type { Instance, Record } from "./vultr";
 
-const api =
-  <T>(method: "post" | "get" | "delete", path: string) =>
-  (data?: Record<string, unknown>): Promise<T> =>
-    new Promise((res, rej) => {
-      axios
-        .request({
-          url: `https://api.vultr.com/v2/${path}`,
-          method,
-          headers: {
-            Authorization: `Bearer ${get("api_key")}`,
-          },
-          data,
-        })
-        .then(({ data }) => res(data))
-        .catch((err) => {
-          console.log(`https://api.vultr.com/v2/${path}`);
-          rej(
-            err.response?.data
-              ? new Error(JSON.stringify(err.response.data))
-              : err
-          );
-        });
-    });
+const PAGINATION_DEFAULT = 100
+const PULL_REQUEST_TAG = 'pullrequest'
+const UNINITIALISED_IP_ADDRESS = '0.0.0.0'
+const VULTR_PLAN = 'vc2-1c-1gb'
+const VULTR_REGION = 'lhr'
 
-export const createInstance = api<Vultr.Instance>("post", "instances");
+const getRecords = async (
+  vultr: any,
+  domain: string,
+  cursor: string | undefined = undefined,
+): Promise<any> => {
+  return await vultr.dns.listRecords({
+    'dns-domain': domain,
+    per_page: PAGINATION_DEFAULT,
+    cursor: cursor,
+  });
+}
 
-export const getInstance = (id: string) =>
-  api<Vultr.Instance>("get", `instances/${id}`);
+export const getAllRecords = async (
+  vultr: any,
+  domain: string,
+): Promise<Record[]> => {
+  let allRecords = [];
+  let cursor = undefined;
+  while (true) {
+    const { records, meta } = await getRecords(vultr, domain, cursor);
+    allRecords.push(...records);
+    cursor = meta.links?.next;
+    if (!cursor) break;
+    console.log(`📜 pagination: collected ${allRecords.length} items - requesting next ${PAGINATION_DEFAULT}`)
+  }
+  console.log(`👀 found ${allRecords.length} records on domain ${domain}`)
+  return allRecords;
+}
 
-export const listInstances = (perPage = 100) =>
-  api<{ instances: Vultr.Instance["instance"][] }>(
-    "get",
-    `instances?per_page=${perPage}`
-  );
+const getInstances = async (
+  vultr: any,
+  cursor: string | undefined = undefined,
+): Promise<any> => {
+  return await vultr.instances.listInstances({
+    region: VULTR_REGION,
+    // the vultr-node client wants a string for per_page here 
```

Comment:
Let's catch up and talk types!

Ideally we'd hope that libraries come with types, otherwise there's a reliable store of them available via https://github.com/DefinitelyTyped/DefinitelyTyped

In this instance, both come up empty so we can create a `.d.ts` file to represent the types (as we understand them from the docs) for the `VultrNode` library.

That should mean that comments like this are redundant, and we can remove a lot of `any` types throughout.

Quick / rough example for this function - 

```ts
// Make new file vultr-action/src/@vultr/vultr-node.d.ts
// Include this in tsconifg.json also

declare module "@vultr/vultr-node" {
  function initialize(options: { apiKey: string }): Vultr;

  interface Vultr {
    instances: {
      listInstances: (
        region: string,
        per_page: number,
        cursor?: string
      ) => Promise<any>;
    };
  }
}
```

This could certainly be handled as a follow up PR 👍

Reply:
@DafyddLlyr thanks for the help with this, I think I've implemented a thorough typing system! please let me know if it makes sense to you :)

Reply:
@jessicamcinchak thankssss! I do see the typescript appeal ;)